proto/v1/rollout_service.proto

```diff
@@ -102,8 +118,426 @@ message Plan {
+      MIGRATE_SDL = 3;
+      // Used for DDL changes using gh-ost.
+      MIGRATE_GHOST = 4;
+      // Used when restoring from a backup (the restored database branched from the original backup).
+      BRANCH = 5;
+      // Used for DML change.
+      DATA = 6;
+    }
+    Type type = 3;
+    // schema_version is parsed from VCS file name.
+    // It is automatically generated in the UI workflow.
+    string schema_version = 4;
+
+    // If RollbackEnabled, build the RollbackSheetID of the task.
+    bool rollback_enabled = 6;
+  }
+
+  message RestoreConfig {
+    // The resource name of the target to restore.
+    // Format: instances/{instance}/databases/{database}
+    string target = 1;
+    // create_database_config is present if the user wants to restore to a new database.
+    optional CreateDatabaseConfig create_database_config = 2;
+    // source determines how to restore the database.
+    // 1. from a backup
+    // 2. from a point in time
+    oneof source {
+      // FIXME: format TBD
+      // Restore from a backup.
+      string backup = 3;
+      // After the PITR operations, the database will be recovered to the state at this time.
+      google.protobuf.Timestamp point_in_time = 4;
+    }
+  }
+}
+
+message ListPlanCheckRunsRequest {
+  // The parent, which owns this collection of plan check runs.
+  // Format: projects/{project}/plans/{plan}
+  string parent = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // The maximum number of plan check runs to return. The service may return fewer than
+  // this value.
+  // If unspecified, at most 50 plans will be returned.
+  // The maximum value is 1000; values above 1000 will be coerced to 1000.
+  int32 page_size = 2;
+
+  // A page token, received from a previous `ListPlanCheckRuns` call.
+  // Provide this to retrieve the subsequent page.
+  //
+  // When paginating, all other parameters provided to `ListPlanCheckRuns` must match
+  // the call that provided the page token.
+  string page_token = 3;
+}
+
+message ListPlanCheckRunsResponse {
+  // The plan check runs from the specified request.
+  repeated PlanCheckRun plan_check_runs = 1;
+
+  // A token, which can be sent as `page_token` to retrieve the next page.
+  // If this field is omitted, there are no subsequent pages.
+  string next_page_token = 2;
+}
+
+message PlanCheckRun {
+  // Format: projects/{project}/plans/{plan}/planCheckRuns/{planCheckRun}
+  string name = 1;
+
+  // The system-assigned, unique identifier for a resource.
+  string uid = 2;
+
+  enum Type {
+    TYPE_UNSPECIFIED = 0;
+    DATABASE_STATEMENT_FAKE_ADVISE = 1;
+    DATABASE_STATEMENT_SYNTAX = 2;
+    DATABASE_STATEMENT_COMPATIBILITY = 3;
+    DATABASE_STATEMENT_ADVISE = 4;
+    DATABASE_STATEMENT_TYPE = 5;
+    DATABASE_STATEMENT_TYPE_REPORT = 6;
+    DATABASE_STATEMENT_AFFECTED_ROWS_REPORT = 7;
+    DATABASE_CONNECT = 8;
+    DATABASE_GHOST_SYNC = 9;
+    DATABASE_PITR_MYSQL = 10;
+  }
+  Type type = 3;
+
+  enum Status {
+    UNKNOWN = 0;
```

Comment:
Should have `STATUS_UNSPECIFIED = 0;`?

Reply:
https://google.aip.dev/126#:~:text=An%20exception%20to%20this%20rule%20is%20if%20there%20is%20a%20clearly%20useful%20zero%20value.%20In%20particular%2C%20if%20an%20enum%20needs%20to%20present%20an%20UNKNOWN%2C%20it%20is%20usually%20clearer%20and%20more%20useful%20for%20it%20to%20be%20a%20zero%20value%20rather%20than%20having%20both.

Reply:
> The first value of the enum should be the name of the enum itself followed by the suffix _UNSPECIFIED.
An exception to this rule is if there is a clearly useful zero value. In particular, if an enum needs to present an UNKNOWN, it is usually clearer and more useful for it to be a zero value rather than having both.